src/config.rs

```diff
@@ -67,7 +81,15 @@ async fn init_config() -> Config {
     // init the queue
     let queue = Box::new(SqsQueue {});
 
-    Config { starknet_client: Arc::new(provider), da_client: build_da_client(), database, queue }
+    let settings_provider = DefaultSettingsProvider {};
```

Comment:
Yes exactly, or YAML, or generally anything (e.g. remote config server).
It helps to avoid multiple config files that potentially duplicate each other, plus hard to manage.
Also it's easier to initialize sub-components -- you can just forward the provider without knowing the actual config type (that's especially useful in our case when we have various backends for the same task).